### PR TITLE
[HDFS-450] Add HDFS upgrade-downgrade integration test

### PR DESCRIPTION
**Pull request by susanxhuynh**
A test that does upgrade / downgrade between the currently released version of beta-hdfs and the version in master branch.

### CODE REVIEW

**Comment by susanxhuynh** [comment]
Will re-open if/when upgrades are fixed.

**Comment by NimaVaziri** [comment]
@susanxhuynh mind taking a look at this updated branch? Can't tag you as a reviewer since you authored the PR

**Comment by NimaVaziri** [comment]
The majority of the `foo` => `sdk_foo` change was merged in https://github.com/mesosphere/dcos-commons/pull/1267 though the remaining portion is cleaned up here as an extension of the `sdk_test_upgrade` => `sdk_versions` modification.

File: frameworks/hdfs/tests/test_upgrade_downgrade.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import sdk_versions
+from tests.config import *
+
+@pytest.mark.upgrade
+@pytest.mark.sanity
+def test_upgrade():
+    options = {
+        "service": {
+            "name": "beta-{}".format(PACKAGE_NAME),
```
Comment (NimaVaziri) [inline frameworks/hdfs/tests/test_upgrade_downgrade.py]: I'll rerun it to confirm but I remember when I actually ran the test without `beta-{}` there ended up being a discrepancy between service names so the polling for deployment and recovery would be polling using the wrong name and as a result would timeout.

Comment (NimaVaziri) [inline frameworks/hdfs/tests/test_upgrade_downgrade.py]: Nevermind, confirmed the beta format isn't need. Reverted to just package name.

Comment (susanxhuynh) [inline frameworks/hdfs/tests/test_upgrade_downgrade.py]: Cool, thanks, @NimaVaziri

**Comment by NimaVaziri** [comment]
Closing this as the upgrade itself is handled in @nickbp's [upgrade PR](https://github.com/mesosphere/dcos-commons/pull/1339) and the journal node readiness check is handled in https://github.com/mesosphere/dcos-commons/pull/1351

**Pull request closed without merging**